### DISCUSSION TITLE
Code': 400, 'message': 'No role ids sent

### DISCUSSION THREAD

**Discussion by Tom-stack-crypto** [CrowdStrike/falconpy#199] [Q&A] [answered]
Hello,

i have just started working with the falconpy and am having a problem with with passing the role id with the GrantUserRoleIds.
What is the correct syntax for the body?

Getting the error as in the title.
```
BODY = {
'id':'remote_responder' 
}
response = falcon.GrantUserRoleIds(parameters=PARAMS, body=BODY)
```

**Comment by jshcodes** [comment, marked as answer]
Hi @Tom-stack-crypto!

There are are two payload elements we need to pass here.  

- The first will be our __user_uuid__, which will be passed as a query parameter.
- The second will be the __roleId__, which we pass as a body parameter. 
    - Multiple role IDs can be assigned at once, so the body payload parameter is passed as a list.

```python
from falconpy.user_management import User_Management
falcon = User_Management(creds={"client_id": "CLIENT_ID_HERE", "client_secret": "CLIENT_SECRET_HERE"})

PARAMS = {"user_uuid": "USER UUID STRING HERE (Check out RetrieveUserUUID)"}
BODY = {
  "roleIds": [
    "vulnerability_manager"  # Or whatever role(s) you wish to assign
  ]
}  

result = falcon.GrantUserRoleIds(parameters=PARAMS, body=BODY)
print(result)
```

Give this a try, and let us know if there are any questions.
